libsolidity/formal/SMTChecker.cpp

```diff
@@ -909,3 +894,16 @@ void SMTChecker::addPathImpliedExpression(smt::Expression const& _e)
 {
 	m_interface->addAssertion(smt::Expression::implies(currentPathConditions(), _e));
 }
+
+void SMTChecker::clearLocalVariables()
+{
+	for (auto it = m_variables.begin(); it != m_variables.end(); )
```

Comment:
Could something like `boost::remove_if` be used to make this a little less error-prone? If it would be much more complicated, then this way is still better, I think.

Reply:
`remove_if` doesn't work on maps.
Anyway this way seems to be ok:
http://en.cppreference.com/w/cpp/experimental/map/erase_if